core/worker.go

```diff
@@ -1515,10 +1528,42 @@ func createCoinbaseTxWithFees(header *types.WorkObject, fees *big.Int, state *st
 		TxIn:  []types.TxIn{in},
 		TxOut: outs,
 	}
+	txDigestHash := signer.Hash(types.NewTx(qiTx))
+	sig, err := schnorr.Sign(ephemeralKey, txDigestHash[:])
+	if err != nil {
+		return nil, err
+	}
+	qiTx.Signature = sig
+	if !sig.Verify(txDigestHash[:], ephemeralKey.PubKey()) {
+		return nil, fmt.Errorf("ephemeral key signature verification failed")
+	}
 	tx := types.NewTx(qiTx)
 	for i, out := range qiTx.TxOut {
 		// this may be unnecessary
 		state.CreateUTXO(tx.Hash(), uint16(i), types.NewUtxoEntry(&out))
 	}
 	return tx, nil
 }
+
+// createQuaiCoinbaseTx returns a coinbase transaction paying an appropriate subsidy
+// based on the passed block height to the provided address.
+func createQuaiCoinbaseTx(state *state.StateDB, header *types.WorkObject, logger *log.Logger, signer types.Signer, key *ecdsa.PrivateKey) (*types.Transaction, error) {
+	// Select the correct block reward based on chain progression
+	blockReward := misc.CalculateReward(header)
+	coinbase := header.Coinbase()
+	internal, err := coinbase.InternalAndQuaiAddress()
+	if err != nil {
+		logger.WithFields(log.Fields{
+			"Address": header.Coinbase().String(),
+			"Hash":    header.Hash().String(),
+		}).Error("Block has out of scope coinbase, skipping block reward")
+		return nil, fmt.Errorf("block has out of scope coinbase, skipping block reward")
+	}
+	tx := types.NewTx(&types.QuaiTx{To: &coinbase, Value: blockReward, Data: common.Hex2Bytes("Quai block reward")})
+	tx, err = types.SignTx(tx, signer, key)
```

Comment:
Curious why you added this back? afaict it is inconsequential, so we removed it to save some space.. but no real objection to having it.

Reply:
It's required to have a signature for transaction hash, because we use recover to get the sender location